GH-34956: [Docs][Python] Add to docs the usage of the FixedShapeTensorType (#34957)
apache/arrow: docs/source/python/extending_types.rst

This way, you can control the conversion of a pyarrow ``Array`` of your pyarrow
extension type to a pandas ``ExtensionArray`` that can be stored in a DataFrame.

Canonical extension types
~~~~~~~~~~~~~~~~~~~~~~~~~

In the :ref:`format_canonical_extensions` under the section **Official List**

> there is a list of canonical extension types. Here you can find an example of how
> to implement and how to use the listed canonical extension types.

Reviewer: Implement seems too strong here? :D
```suggestion
there is a list of canonical extension types. Here we list examples of how to
use the listed canonical extension types.
```
